File: LeetCode/DataStruct/JS/Leetcode856-score-of-parentheses.js
var scoreOfParentheses = function(s) {
  let n = s.length;
  let stk = [0];
  for(let i = 0;i < n;i++) {
    if(s[i] == '(') stk.push(0);
    else {
      let t = stk.pop();
      if(t == 0) {
        stk[stk.length - 1] += 1;
      }else {
        stk[stk.length - 1] += 2 * t;
      }
    }
  }
  return stk.pop()
};

File: LeetCode/DataStruct/Rust/Leetcode856-score-of-parentheses.rs
impl Solution {
  pub fn score_of_parentheses(s: String) -> i32 {
    let (n, s) = (s.len(), s.as_bytes());
    let mut stk = vec![0];
    for i in 0..n {
      if s[i] == b'(' {
        stk.push(0);
      } else {
        let t = stk.pop().unwrap();
        let last = stk.len();
        if t == 0 {
          stk[last - 1] += 1;
        } else {
          stk[last - 1] += 2 * t;
        }
      }
    }
    stk[0]
  }
}
